src/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt

```diff
@@ -43,6 +43,17 @@ set(NATIVECRYPTO_SOURCES
     pal_x509ext.cpp
 )
 
+if (FEATURE_DISTRO_AGNOSTIC_SSL)
+    if (APPLE OR (CMAKE_SYSTEM_NAME STREQUAL FreeBSD) OR (CMAKE_SYSTEM_NAME STREQUAL NetBSD))
```

Comment:
Should we change the condition to fail except on Linux,  rather  than listing out those it fails on? Other systems could be opted in over time.